astropy/io/fits/fitsrec.py

```diff
@@ -553,11 +550,13 @@ def __setitem__(self, key, value):
             raise TypeError('Assignment requires a FITS_record, tuple, or '
                             'list as input.')
 
-    def __getslice__(self, start, end):
-        return self[slice(start, end)]
+    if PY2:
```

Comment:
The test here should perhaps be `if hasattr(ndarray, '__getslice__')`, to also not define these in newer version of numpy